scheduler/compileserver.cpp

```diff
@@ -401,3 +403,52 @@ bool CompileServer::blacklisted(const Job *job, const pair<string, string> &envi
     Environments blacklist = job->submitter()->getEnvsForBlacklistedCS(this);
     return find(blacklist.begin(), blacklist.end(), environment) != blacklist.end();
 }
+
+bool CompileServer::incomingConnectionTestRequired() const
+{
+    if(m_noRemote || (time(0) < m_nextConnTime))
+    {
+        return false;
+    }
+
+    return true;
+}
+
+bool CompileServer::incomingConnectionAccepted()
+{
+#if DEBUG_SCHEDULER > 0
+        trace()  << "Verifying  " << nodeName() << " is accepting incoming connections" << endl;
+#endif
+    //Time between retries are doubled each failed attempt(seconds)
+    //The delay between retry attempts are capped at 4096s (~1 hr)
+    static const size_t table_size = 12;
+    static const time_t time_offset_table[table_size] = {
+           2,    4,    8,    16,    32,
+          64,  128,  256,   512,  1024,
+        2048, 4096
+    };
+
```

Comment:
This table contains the min time between failed connection retries(does not include the timeout (5s) and other blocking code in the main loop). The numbers are generated to double the time each failure but we could get a less drastic increases by using something like e^(-n/a) +1 which would give us values 2 < f(n) < 1 which we can use to multiply against f(n-1) time.